File: Exercicios_diarios/Revisão_intermediaria/Localizar índice de pivô.cpp
class Solution
{
public:
    int pivotIndex(vector<int> &nums)
    {
        int direita = 0, esquerda = 0;
        for (int &n : nums) // equivaentea ao for n in nums em 
        {
            direita += n;
        }
        for (int i = 0; i < nums.size(); i++)
        {
            if (esquerda == direita - esquerda - nums[i])
            {
                return i;
            }
            esquerda += nums[i];
        }
        return -1;
    }
};

File: Python/Exercicios_diarios/Revisão_intermediaria/Localizar índice de pivô.py
class Solution:
    def pivotIndex(self, nums: list[int]) -> int:
        total, esquerda, direita = sum(nums), 0, 0
        for i in range(len(nums)):
            direita = total-esquerda-nums[i]
            if direita == esquerda:
                return i
            esquerda += nums[i]
        return -1
